core/app/crash/crash.go

```diff
@@ -19,20 +19,33 @@
 package crash
```

Comment:
Please don't change this package. Just add another method to the `crash/reporting` package to deal with external process crashes.

Reply:
What about the shutdown function setter? Without this change the cancelFunc set in app.handleCrashSignals could preempt any other crash handlers registered and shut down the app before they can report anything.

Reply:
> Without this change the cancelFunc set in app.handleCrashSignals could preempt any other crash handlers registered and shut down the app before they can report anything.

Nothing in the crash handler should depend on the context cancellation state. Are you seeing this somewhere?